### DISCUSSION TITLE
Berry UDP class, udp.begin not binding to address (ETH module)

### DISCUSSION THREAD

**Discussion by photocromax** [arendst/Tasmota#17402] [Q&A] [answered]
Hello,

I'm having a issue with a WT32-ETH01 (v1.4)
I noticed, since V12.2.0.5 (or .6), berry udp.begin method was always binding to wifi interface and not the eth interface as specified in the code.

I waited the problem could be solved with tasmota V12.3.0.1, but it still there, so I looked for the source code.

It seems the parameter "host" is completely disregarded.

V12.3 lib/libesp32/berry_tasmota/src/be_udp_lib.cpp
```
 // udp.begin(address:string, port:int) -> bool
  int32_t be_udp_begin_ntv(WiFiUDP *udp, int32_t port) {   
    IPAddress addr;
    // AddLog(LOG_LEVEL_DEBUG, "BRY: udp.begin listening to '%s'", addr.toString().c_str());
    return udp->begin(addr, port);
  }
```

V12.2
```
 // udp.begin(address:string, port:int) -> nil
  int32_t be_udp_begin_ntv(WiFiUDP *udp, const char *host, int32_t port) {
    IPAddress addr((uint32_t)0);
    // if no host or host is "" then we defult to INADDR_ANY (0.0.0.0)
    if(host && (*host != 0) && !WiFiGenericClass::hostByName(host, addr)){
        return 0;
    }
    return udp->begin(addr, port);
  }
```
Diff link:
[lib/libesp32/berry_tasmota/src/be_udp_lib.cpp](https://github.com/arendst/Tasmota/compare/v12.2.0...v12.3.0#diff-ff2a1a13abf9b027e2eb9627859d319753cba983ae212cabef766afcda02b8ea)

Even if binding on all interfaces, when I try to send udp packets  via the ethernet interface, the packets are just sent via the wifi.

Is this a new limitation or just an error?

Thank you for your attention

**Comment by s-hadinger** [comment, marked as answer]
Oops. It was meant to be a simplification and I was mainly thinking about receiving packets. I will revert to the previous behavior.

**Comment by photocromax** [comment]
Thanks,

is it easier to modify udp.send() adding the sender address parameter?

In that case it wouldn't be needed to begin a dedicated listener.
